src/shared/label-util.c

```diff
@@ -115,3 +116,29 @@ int btrfs_subvol_make_label(const char *path) {
 
         return mac_smack_fix(path, 0);
 }
+
+static int label_pre(int dir_fd, const char *path, mode_t mode) {
+        return mac_selinux_create_file_prepare_at(dir_fd, path, mode);
+}
+
+static int label_post(int dir_fd, const char *path) {
+        mac_selinux_create_file_clear();
+        return mac_smack_fix_full(dir_fd, path, NULL, 0);
+}
+
+static const LabelOps *label_ops = &(LabelOps) {
+        .pre = label_pre,
+        .post = label_post,
+};
+
+int mac_init(void) {
+        int r;
+
+        r = mac_selinux_init();
+        if (r < 0)
+                return r;
+
+        label_ops_set(label_ops);
```

Comment:
shouldn't this be done only if selinux or smack are enabled?

Reply:
The labeling functions are currently called unconditionally and return early if selinux/smack is not enabled, so we can safely always set the labelling ops here.